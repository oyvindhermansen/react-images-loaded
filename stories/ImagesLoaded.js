import React from 'react';
import { storiesOf } from '@storybook/react';
import FullPackage from './variants/FullPackage';

storiesOf('ImagesLoaded', module).add('Full package', () => <FullPackage />);
