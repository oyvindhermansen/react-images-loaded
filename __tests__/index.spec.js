import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';
import ImagesLoaded from '../src/index';
import renderer from 'react-test-renderer';

describe('<ImagesLoaded />', () => {
  it('should render and work.', async () => {
    const done = jest.fn();
    const onAlways = jest.fn();
    const onFail = jest.fn();
    const onProgress = jest.fn();
    const onUpdate = jest.fn();

    const component = renderer.create(
      <ImagesLoaded
        done={done}
        onAlways={onAlways}
        onFail={onFail}
        onProgress={onProgress}
        onUpdate={onUpdate}
      >
        {['img1.jpg'].map((item, index) => <img key={index} src={item} />)}
      </ImagesLoaded>
    );

    expect(component.toJSON()).toMatchSnapshot();

    const instance = component.getInstance();
    const cWU = jest.spyOn(instance, 'componentWillUnmount');
    const cDU = jest.spyOn(instance, 'componentDidUpdate');

    component.update(
      <ImagesLoaded
        done={done}
        onAlways={onAlways}
        onFail={onFail}
        onProgress={onProgress}
        onUpdate={onUpdate}
      >
        {['img1.jpg', 'img2.jpg'].map((item, index) => (
          <img key={index} src={item} />
        ))}
      </ImagesLoaded>
    );

    expect(cDU).toHaveBeenCalled();

    component.unmount();

    expect(cWU).toHaveBeenCalled();
  });

  it('should render children', () => {
    const testChildren = ['div', 'section', 'img'];
    const component = ReactTestUtils.renderIntoDocument(
      <ImagesLoaded>
        {testChildren.map((item, index) => {
          return <div key={index} className={'child'} />;
        })}
      </ImagesLoaded>
    );

    const children = ReactTestUtils.scryRenderedDOMComponentsWithClass(
      component,
      'child'
    );

    expect(children.length).toEqual(3);
  });

  it('should set the correct default props', () => {
    const component = ReactTestUtils.renderIntoDocument(<ImagesLoaded />);

    const defaultProps = {
      className: 'images-loaded-container',
      elementType: 'div',
      background: false
    };

    expect(component.props).toEqual(defaultProps);
  });

  it('should render with the correct elementType', () => {
    const defaultComponent = ReactTestUtils.renderIntoDocument(
      <ImagesLoaded />
    );

    const componentWithList = ReactTestUtils.renderIntoDocument(
      <ImagesLoaded elementType={'ul'} />
    );

    expect(
      ReactTestUtils.scryRenderedDOMComponentsWithTag(defaultComponent, 'div')
        .length
    ).toBeTruthy();

    expect(
      ReactTestUtils.scryRenderedDOMComponentsWithTag(componentWithList, 'ul')
        .length
    ).toBeTruthy();

    expect(
      ReactTestUtils.scryRenderedDOMComponentsWithTag(
        componentWithList,
        'section'
      ).length
    ).toBeFalsy();
  });
});
