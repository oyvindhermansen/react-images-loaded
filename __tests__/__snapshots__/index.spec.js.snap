// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImagesLoaded /> should render and work. 1`] = `
<div
  className="images-loaded-container"
>
  <img
    src="img1.jpg"
  />
</div>
`;
